### ISSUE TITLE
aws_batch.JobDefinition.job_definition_name returns arn not name

### ISSUE THREAD

**Issue by bencaldwell** [aws/aws-cdk#7069]
I am running cdk synth in a codebuild step, then CloudFormationCreateUpdateStackAction to deploy the template. My deployment is failing when I reference the property `aws_batch.JobDefinition.job_definition_name` as it returns the *arn* of the job, not the job definition name as expected..

### Reproduction Steps

```
job = aws_batch.JobDefinition(
            self,
            id="job-definition",
            job_definition_name="job-definition-name",
            container=aws_batch.JobDefinitionContainer(
                memory_limit_mib=12024,
                privileged=False,
                job_role=job_role,
                read_only=False,
                vcpus=1,
                image=aws_ecs.ContainerImage.from_ecr_repository(
                    repository=aws_ecr.Repository.from_repository_name(
                        self,
                        id="image-repo",
                        repository_name=constant.IMAGE_REPO
                    ),
                    tag="LATEST"
                ),
                command=[
                    "python",
                    "--version"
                ]
            ),
        )

job.job_definition_arn #prints the arn
job.job_definition_name #prints the arn too - should print "job-definition-name"
```

### Environment

  - **CLI Version      :** aws-cli/1.14.44 Python/3.6.9 Linux/5.3.0-42-generic botocore/1.8.48
  - **Framework Version:** 1.31.0
  - **OS               :** aws_codebuild.LinuxBuildImage.STANDARD_3_0
  - **Language         :** Python

### Other

--- 

This is :bug: Bug Report

**Comment by iliapolo** [comment]
Hi @bencaldwell 

While this is a bit confusing, its actually behaving as expected. Let me explain.

When you write `job.job_definition_*`, you are accessing runtime attributes of the resource, and **not** the configuration properties you passed in the initialization. This is intentional and allows extracting dynamic information that is determined only by CloudFormation post resource creation (or in CDK world - post initialization of the construct).

In our case, the `job_definition_name` you passed in the constructor corresponds to the config property when creating a job definition, but `job.job_definition_name` corresponds to the `JobDefinition` **resource name that is returned by CloudFormation**.

The value returned by CloudFormation is determined by the [`Ref`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-batch-jobdefinition.html) return value of the spec. Notice it says it will return the `Arn` of the resource.

In other cases, for example for [Buckets](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html), CloudFormation will return the bucket name, and the `Arn` is fetched with a different attribute called `Arn`. This does not exist in `JobDefinition`, meaning that both `arn` and `name` will return the same thing.

Bottom line, CloudFormation simply doesn't have the distinction between the two for this resource.

Was that somewhat clear? :)

Having said all that, I agree this is a case of unfortunate naming that leads to confusion, and perhaps can be mitigated with a little renaming.

**Comment by ztane** [comment]
If only the **documentation** would actually *say* anything towards that, but **no**.
